include/dlaf/util_matrix.h

```diff
@@ -387,6 +387,45 @@ void set_random_hermitian_positive_definite(Matrix<T, Device::CPU>& matrix) {
   internal::set_random_hermitian_with_offset(matrix, 2 * matrix.size().rows());
 }
 
+/// The tiles are returned in column major order
+template <class T, Device D>
+std::vector<pika::future<matrix::Tile<T, D>>> collectReadWriteTiles(GlobalTileIndex begin,
+                                                                    GlobalTileSize sz,
+                                                                    Matrix<T, D>& mat) {
+  std::vector<pika::future<matrix::Tile<T, D>>> tiles;
+  tiles.reserve(to_sizet(sz.linear_size()));
+  for (auto idx : iterate_range2d(begin, sz)) {
+    tiles.push_back(mat(idx));
+  }
+  return tiles;
+}
+
+/// The tiles are returned in column major order
+template <class T, Device D>
+std::vector<pika::shared_future<matrix::Tile<const T, D>>> collectReadTiles(GlobalTileIndex begin,
+                                                                            GlobalTileSize sz,
+                                                                            Matrix<const T, D>& mat) {
+  std::vector<pika::shared_future<matrix::Tile<const T, D>>> tiles;
+  tiles.reserve(to_sizet(sz.linear_size()));
+  for (auto idx : iterate_range2d(begin, sz)) {
+    tiles.push_back(mat.read(idx));
+  }
+  return tiles;
+}
+
+/// The tiles are returned in column major order
+template <class T, Device D>
+auto collectReadTileSenders(GlobalTileIndex begin, GlobalTileSize sz, Matrix<const T, D>& mat) {
+  using pika::execution::experimental::keep_future;
+
+  std::vector<decltype(keep_future(pika::shared_future<matrix::Tile<const T, D>>()))> tiles;
+  tiles.reserve(to_sizet(sz.linear_size()));
+  for (auto idx : iterate_range2d(begin, sz)) {
+    tiles.push_back(mat.read_sender(idx));
+  }
+  return tiles;
+}
```

Comment:
It looks to me like you don't need the non-sender versions of these, so I would just make the "default" use `Matrix::read_sender` and `Matrix::readwrite_sender`. For read-write `Matrix::operator()` and `Matrix::readwrite_sender` are equivalent, but I'm trying to phase out the non-sender accesses (and then we can eventually just remove the `_sender` members).
```suggestion
template <class T, Device D>
auto collectReadWriteTile(GlobalTileIndex begin,
                           GlobalTileSize sz,
                           Matrix<T, D>& mat) {
  std::vector<decltype(mat.readwrite_sender(std::declval<LocalTileIndex>()))> tiles;
  tiles.reserve(to_sizet(sz.linear_size()));
  for (auto idx : iterate_range2d(begin, sz)) {
    tiles.push_back(mat(idx));
  }
  return tiles;
}

/// The tiles are returned in column major order
template <class T, Device D>
auto collectReadTile(GlobalTileIndex begin, GlobalTileSize sz, Matrix<const T, D>& mat) {
  std::vector<decltype(mat.read_sender(std::declval<LocalTileIndex>()))> tiles;
  tiles.reserve(to_sizet(sz.linear_size()));
  for (auto idx : iterate_range2d(begin, sz)) {
    tiles.push_back(mat.read_sender(idx));
  }
  return tiles;
}
```